### PR TITLE
Fix: Fix bugs 2293 & 2329 & 2296 

### PR DESCRIPTION
**Pull request by wen-qing-255**

### CODE REVIEW

File: src/app/flux/manager/DefinitionManager.js [hunk 1]
```diff
@@ -377,12 +383,20 @@ class DefinitionManager {
             definition.settings.wall_line_width_0.default_value = extruderLDefinition.settings.wall_line_width_0.default_value;
             definition.settings.wall_line_width_x.default_value = extruderLDefinition.settings.wall_line_width_x.default_value;
             definition.settings.skin_line_width.default_value = extruderLDefinition.settings.skin_line_width.default_value;
+            definition.settings.wall_material_flow.default_value = extruderLDefinition.settings.material_flow.default_value;
+            definition.settings.skin_material_flow.default_value = extruderLDefinition.settings.material_flow.default_value;
+            definition.settings.roofing_material_flow.default_value = extruderLDefinition.settings.material_flow.default_value;
+            definition.settings.infill_material_flow.default_value = extruderLDefinition.settings.material_flow.default_value;
```
Comment (jane-rose) [inline src/app/flux/manager/DefinitionManager.js:389]: why `infill_material_flow` rely on `infill` and  `shell`

File: src/app/ui/views/ProfileManager/ConfigValueBox.jsx [hunk 2]
```diff
@@ -91,7 +91,7 @@ function ConfigValueBox({ optionConfigGroup, calculateTextIndex, isCategorySelec
                                     )}
                                     {/* eslint no-return-assign: 0*/}
                                     <div className={`${managerType === HEAD_CNC && group.name === 'Carving Tool' && 'sm-flex justify-space-between'}`}>
-                                        <div ref={el => fieldsDom.current[index] = el}>
+                                        <div ref={el => fieldsDom.current[index] = el} className={`${managerType === HEAD_CNC && group.name === 'Carving Tool' && 'width-percent-100'}`}>
```
Comment (jane-rose) [inline src/app/ui/views/ProfileManager/ConfigValueBox.jsx:94]: why add `group.name === 'Carving Tool'`

File: src/app/ui/widgets/PrintingVisualizer/VisualizerPreviewControl.jsx [hunk 3]
```diff
@@ -40,33 +40,30 @@ function useShowToggleBtn() {
     };
 }
 
-const MAX = 288;
-const MIN = 30;
+// const MAX = 288;
+// const MIN = 30;
```
Comment (jane-rose) [inline src/app/ui/widgets/PrintingVisualizer/VisualizerPreviewControl.jsx:44]: remove log

File: src/app/ui/views/ProfileManager/ConfigValueBox.jsx (hunk 2 above)
Comment (wen-qing-255) [reply src/app/ui/views/ProfileManager/ConfigValueBox.jsx:94]: because only Carving Tool need to show tool image

Comment (xiekun1992) [reply src/app/ui/views/ProfileManager/ConfigValueBox.jsx:94]: the className may be `false` ?

**Review by xiekun1992 — APPROVED** [review]

**Pull request merged**